stdlib/src/csv/csv.mojo

```diff
@@ -0,0 +1,252 @@
+        csvfile: FileHandle,
+        delimiter: String,
+        quotechar: String = '"',
+        escapechar: String = "",
+        doublequote: Bool = False,
+        skipinitialspace: Bool = False,
+        lineterminator: String = "\r\n",
+        quoting: Int = QUOTE_MINIMAL,
+    ) raises:
+        """
+        Initialize a Dialect object.
+
+        Args:
+            csvfile: The CSV file to read from.
+            delimiter: The delimiter used to separate fields.
+            quotechar: The character used to quote fields containing special
+                characters.
+            escapechar: The character used to escape the delimiter or quotechar.
+            doublequote: Whether quotechar inside a field is doubled.
+            skipinitialspace: Whether whitespace immediately following the
+                delimiter is ignored.
+            lineterminator: The sequence used to terminate lines.
+            quoting: The quoting mode.
+        """
+        self._dialect = Dialect(
+            delimiter=delimiter,
+            quotechar=quotechar,
+            escapechar=escapechar,
+            doublequote=doublequote,
+            skipinitialspace=skipinitialspace,
+            lineterminator=lineterminator,
+            quoting=quoting,
+        )
+        self._dialect.validate()
+
+        # TODO: Implement streaming to prevent loading the entire file into memory
+        self._lines = csvfile.read().splitlines()
+
+    fn get_line(self: Self, idx: Int) raises -> List[String]:
+        """
+        Returns an specific line in the CSV file.
+
+        Args:
+            idx: The index of the line to return.
+
+        Returns:
+            The line at the given index.
+        """
+        var line: List[String] = self._lines[idx].split(self._dialect.delimiter)
+        for field_ref in line:
+            if (
+                len(field_ref[]) > 1
+                and field_ref[][0] == self._dialect.quotechar
+                and field_ref[][-1] == self._dialect.quotechar
+            ):
+                field_ref[] = field_ref[][1:-1]
+            # TODO: Handle the doublequote, skipinitialspace and escapechar options
+        return line
+
+    fn lines_count(self: Self) -> Int:
+        """
+        Returns the number of lines in the CSV file.
+
+        Returns:
+            The number of lines in the CSV file.
+        """
+        # TODO: This has no sense once we implement streaming
+        return len(self._lines)
+
+    fn __iter__(self: Self) raises -> _ReaderIter[__lifetime_of(self)]:
+        """
+        Iterate through the CSV lines.
+
+        Returns:
+            Iterator.
+        """
+        return _ReaderIter[__lifetime_of(self)](reader_ref=self, idx=0)
+
+
+# ===------------------------------------------------------------------=== #
+# Auxiliary functions
+# ===------------------------------------------------------------------=== #
+
+
+fn _validate_dialect(dialect: Dialect) raises -> Bool:
```

Comment:
```suggestion
fn _validate_reader_dialect(dialect: Dialect) raises -> Bool:
```

Just picked up that `writer` and `reader` might differ in `Dialect` rules, see [here](https://docs.python.org/3/library/csv.html#csv.QUOTE_NONE) and [here](https://docs.python.org/3/library/csv.html#csv.Dialect.escapechar) Do you agree? If so, perhaps set this specific to `reader` for future compatibility?

Reply:
Agreed. Done: https://github.com/msaelices/mojo/commit/46c70208274fa7922e17f0f634b85d12d9e18020

Thanks for the suggestion.